pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -38,6 +38,11 @@ func NewFlowRecords() *FlowRecords {
 
 // AddOrUpdateFlowRecord adds or updates the flow record in the record map given the connection.
 func (fr *FlowRecords) AddOrUpdateFlowRecord(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
+	// if the connection is in dying state and corresponding records are exported,
```

Comment:
Suggestion
"If the connection is in dying state and the corresponding final record is already exported, then there is no need to add the record again"